test/integration/pack/test_pack.py

```diff
@@ -1480,3 +1480,203 @@ def test_overwritten_version_file(cartridge_cmd, project_without_dependencies, t
+    }
+
+    cmd = [
+        cartridge_cmd,
+        "pack", pack_format,
+        project.path,
+    ]
+
+    if platform.system() == 'Darwin':
+        cmd.append('--use-docker')
+
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 0
+
+    archive_path = find_archive(tmpdir, project.name, pack_format)
+    extract_dir = os.path.join(tmpdir, 'extract')
+    extract_app_files(archive_path, pack_format, extract_dir)
+
+    for unit in files_by_units.keys():
+        filename = files_by_units[unit][0]
+        fd_limit = files_by_units[unit][1]
+        filepath = os.path.join(extract_dir, 'etc/systemd/system', filename)
+        with open(filepath) as f:
+            assert "LimitNOFILE={}".format(fd_limit) in f.read()
+
+
+@pytest.mark.parametrize('pack_format', ['rpm', 'deb'])
+def test_fd_limit_file_not_exist(cartridge_cmd, project_without_dependencies, pack_format, tmpdir):
+    project = project_without_dependencies
+
+    not_exist_file_path = "not_exist_file_path.yml"
+
+    cmd = [
+        cartridge_cmd,
+        "pack", pack_format,
+        "--unit-params-file", not_exist_file_path,
+        project.path,
+    ]
+
+    if platform.system() == 'Darwin':
+        cmd.append('--use-docker')
+
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 1
+    assert f"Specified file with system unit params {not_exist_file_path} doesn't exists" in output
+
+
+@pytest.mark.parametrize('pack_format', ['rpm', 'deb'])
+def test_fd_limit_invalid_values(cartridge_cmd, project_without_dependencies, pack_format, tmpdir):
+    project = project_without_dependencies
+
+    fd_limit = -1
+
+    systemd_unit_params = os.path.join(tmpdir, "systemd-unit-params.yml")
+    with open(systemd_unit_params, "w") as f:
+        f.write(f"""
+                fd-limit: {fd_limit}
+                """)
+
+    replace_project_file(project, 'systemd-unit-params.yml', systemd_unit_params)
+
+    cmd = [
+        cartridge_cmd,
+        "pack", pack_format,
+        project.path,
+    ]
+
+    if platform.system() == 'Darwin':
+        cmd.append('--use-docker')
+
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 1
+    assert "Incorrect value for fd-limit: minimal value is 1024" in output
+
+    stateboard_fd_limit = -2
+
+    systemd_unit_params = os.path.join(tmpdir, "systemd-unit-params.yml")
+    with open(systemd_unit_params, "w") as f:
+        f.write(f"""
+                stateboard-fd-limit: {stateboard_fd_limit}
+                """)
+
+    replace_project_file(project, 'systemd-unit-params.yml', systemd_unit_params)
+
+    cmd = [
+        cartridge_cmd,
+        "pack", pack_format,
+        project.path,
+    ]
+
+    if platform.system() == 'Darwin':
+        cmd.append('--use-docker')
+
+    rc, output = run_command_and_get_output(cmd, cwd=tmpdir)
+    assert rc == 1
+    assert "Incorrect value for stateboard-fd-limit: minimal value is 1024" in output
```

Comment:
Add a test when there is no `systemd-unit-params.yml` in the project root to check that pack wouldn't fail.